packages/typescript-estree/src/convert.ts

```diff
@@ -3107,14 +3108,38 @@ export class Converter {
       return;
     }
 
+    // typescript<5.0.0
     if (nodeHasIllegalDecorators(node)) {
       this.#throwError(
         node.illegalDecorators[0],
         'Decorators are not valid here.',
       );
     }
 
-    for (const modifier of getModifiers(node) ?? []) {
+    // @ts-expect-error -- this is safe as it's guarded
+    const modifiers: ts.ModifierLike[] = node.modifiers;
```

Comment:
instead of using `.modifiers` can we use [`getModifiers`](https://github.com/typescript-eslint/typescript-eslint/blob/aa537e55137d854068bea7c9a1520c2763ea8da3/packages/typescript-estree/src/getModifiers.ts#L7-L31) here?

Reply:
`getModifiers` only returns `ts.Modifier`, but we need `ts.Decorator`s, unless we can `getDecorators` again.

Reply:
I know.

> unless we can getDecorators again.


But I checked both modifiers and decorators here, and `modifiers + decorators` is `node.modifiers`

Reply:
I'm not quite sure I understand - your code below does "for each modifier, only action the decorators"

So couldn't the code be
```ts
for (const decorator of getDecorators(node)) {
  // your code
}

for (const modifier of getModifiers(node)) {
  // old code
}
```

Reply:
Ideally, it should work, but I still can't get why you want 4 loops when you can do it in 1 loop.

Anyway, I tried, but turns out `getModifiers() + getDecorators()` not equals to `node.modifiers`.
[`ts.canHaveModifiers()`](https://github.com/typescript-eslint/typescript-eslint/blob/4bc75716681f2118c752df9360a1c5e97409ea93/packages/typescript-estree/src/getModifiers.ts#L16) and [`ts.canHaveDecorators()`](https://github.com/typescript-eslint/typescript-eslint/blob/4bc75716681f2118c752df9360a1c5e97409ea93/packages/typescript-estree/src/getModifiers.ts#L42) are stopping me to access modifiers on some node.

Should we
1. Remove those check
2. Add option to skip check

?

Reply:
The problem here is that they purposely removed `node.modifiers` from the types in TSv5 - so we definitely don't want to be accessing it directly if we can avoid it as it's clear that their intention is for it to be a truly private property.

I guess we could add an option to skip the checks?